crates/trie/sparse-parallel/src/trie.rs

```diff
@@ -282,6 +285,47 @@ impl ParallelSparseTrie {
         }
     }
 
+    /// Used by `remove_leaf` to ensure that when a node is removed from a lower subtrie that any
+    /// externalities are handled. These can include:
+    /// - Removing the lower subtrie completely, if it is now empty.
+    /// - Updating the `path` field of the lower subtrie to indicate that its root node has changed.
+    ///
+    /// This method assumes that the caller will deal with putting all other nodes in the trie into
+    /// a consistent state after the removal of this one.
+    ///
+    /// ## Panics
+    ///
+    /// - If the removed node was not a leaf or extension.
+    fn remove_node(&mut self, path: &Nibbles) {
```

Comment:
can we make it accept a subtrie as an argument, so that we don't call `subtrie_for_path` twice in cases when we already have the subtrie available? 

https://github.com/paradigmxyz/reth/blob/2819ab2c0e32d546996d2c7534901cfe68ad6c16/crates/trie/sparse-parallel/src/trie.rs#L486 https://github.com/paradigmxyz/reth/blob/2819ab2c0e32d546996d2c7534901cfe68ad6c16/crates/trie/sparse-parallel/src/trie.rs#L569